streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -107,11 +107,11 @@ public void onCompletion(final RecordMetadata metadata, final Exception exceptio
                         } else {
                             if (sendException == null) {
                                 sendException = exception;
-                                if (sendException instanceof ProducerFencedException) {
-                                    log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and it will be closed as it is a zombie.", logPrefix, topic, exception);
-                                } else {
-                                    log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and the exception will eventually be thrown", logPrefix, topic, exception);
-                                }
+                                log.error("{} Error sending record with key {} to topic {}, partition {}. " +
```

Comment:
Why do you remove the if-else? For ProducerFencedException, we will just close the task but not rethrow. Thus,  `and the exception will eventually be thrown` is not true for this case